sonar-db/src/main/java/org/sonar/db/component/ProjectMeasuresIndexerIterator.java

```diff
@@ -64,31 +67,68 @@
     "AND (pm.value IS NOT NULL OR pm.variation_value_1 IS NOT NULL OR pm.text_value IS NOT NULL) " +
     "AND pm.person_id IS NULL ";
 
-  private final DbSession dbSession;
+  private final PreparedStatement measuresStatement;
   private final Map<Long, String> metrics;
+  private final Iterator<Project> projects;
 
-  private ProjectMeasuresResultSetIterator(PreparedStatement stmt, DbSession dbSession, Map<Long, String> metrics) throws SQLException {
-    super(stmt);
-    this.dbSession = dbSession;
+  private ProjectMeasuresIndexerIterator(PreparedStatement measuresStatement, Map<Long, String> metrics, List<Project> projects) throws SQLException {
+    this.measuresStatement = measuresStatement;
     this.metrics = metrics;
+    this.projects = projects.iterator();
   }
 
-  static ProjectMeasuresResultSetIterator create(DbClient dbClient, DbSession session, long afterDate, @Nullable String projectUuid) {
+  public static ProjectMeasuresIndexerIterator create(DbSession session, long afterDate, @Nullable String projectUuid) {
     try {
-      PreparedStatement projectsStatement = createProjectsStatement(dbClient, session, afterDate, projectUuid);
-      Map<Long, String> metricIds = selectMetricIds(session);
-      return new ProjectMeasuresResultSetIterator(projectsStatement, session, metricIds);
+      Map<Long, String> metrics = selectMetricIds(session);
```

Comment:
moreover it should not blacklist metric types but it should whitelist numeric types.